internal/cmd/version.go

```diff
@@ -18,45 +18,59 @@ import (
 )
 
+func getClientVersion(cmd *cobra.Command) string {
+	includeDeps := cobrautil.MustGetBool(cmd, "include-deps")
+	return cobrautil.UsageVersion("zed", includeDeps)
+}
+
+func getServerVersion(cmd *cobra.Command, spiceClient v1.SchemaServiceClient) (string, error) {
+	var headerMD, trailerMD metadata.MD
+	// NOTE: we ignore the error here, as it may be due to a schema not existing, or
+	// the client being unable to connect, etc. We just treat all such cases as an unknown
+	// version.
+	_, _ = spiceClient.ReadSchema(cmd.Context(), &v1.ReadSchemaRequest{}, grpc.Header(&headerMD), grpc.Trailer(&trailerMD))
+	versionFromHeader := headerMD.Get(string(responsemeta.ServerVersion))
+	versionFromTrailer := trailerMD.Get(string(responsemeta.ServerVersion))
+	if len(versionFromHeader) == 1 && len(versionFromTrailer) == 1 && versionFromHeader[0] != versionFromTrailer[0] {
+		return "", fmt.Errorf("mismatched server versions in header (%s) and trailer (%s)", versionFromHeader[0], versionFromTrailer[0])
+	}
+
+	if len(versionFromHeader) == 1 && versionFromHeader[0] != "" {
+		return versionFromHeader[0], nil
+	}
+	if len(versionFromTrailer) == 1 && versionFromTrailer[0] != "" {
+		return versionFromTrailer[0], nil
+	}
+	return "(unknown)", nil
+}
+
 func versionCmdFunc(cmd *cobra.Command, _ []string) error {
 	if !isatty.IsTerminal(os.Stdout.Fd()) {
 		color.Disable()
 	}
 
+	green := color.FgGreen.Render
+	fmt.Print(green("client: "))
+	console.Println(getClientVersion(cmd))
+
 	includeRemoteVersion := cobrautil.MustGetBool(cmd, "include-remote-version")
-	hasContext := false
 	if includeRemoteVersion {
 		configStore, secretStore := client.DefaultStorage()
 		_, err := client.GetCurrentTokenWithCLIOverride(cmd, configStore, secretStore)
-		hasContext = err == nil
-	}
-
-	if hasContext && includeRemoteVersion {
-		green := color.FgGreen.Render
-		fmt.Print(green("client: "))
-	}
-
-	console.Println(cobrautil.UsageVersion("zed", cobrautil.MustGetBool(cmd, "include-deps")))
-
-	if hasContext && includeRemoteVersion {
-		client, err := client.NewClient(cmd)
-		if err != nil {
-			return err
-		}
 
-		// NOTE: we ignore the error here, as it may be due to a schema not existing, or
-		// the client being unable to connect, etc. We just treat all such cases as an unknown
-		// version.
-		var headerMD metadata.MD
-		_, _ = client.ReadSchema(cmd.Context(), &v1.ReadSchemaRequest{}, grpc.Header(&headerMD))
-		version := headerMD.Get(string(responsemeta.ServerVersion))
+		if err == nil {
+			spiceClient, err := client.NewClient(cmd)
+			if err != nil {
+				return err
+			}
+			serverVersion, err := getServerVersion(cmd, spiceClient)
+			if err != nil {
+				return err
+			}
 
-		blue := color.FgLightBlue.Render
-		fmt.Print(blue("service: "))
-		if len(version) == 1 {
-			console.Println(version[0])
-		} else {
-			console.Println("(unknown)")
+			blue := color.FgLightBlue.Render
+			fmt.Print(blue("service: "))
+			console.Println(serverVersion)
 		}
 	}
 
```

Comment:
I refactored this code because of this warning that my IDE gave me. I think I am preserving behavior.

<img width="1144" height="676" alt="image" src="https://github.com/user-attachments/assets/5afad412-fac1-415c-85c3-4e196eaaea25" />

Reply:
Why would it always be true? 🤔

Reply:
Or is it `hasContext` that's always true by the time you hit this branch?

Reply:
nvm I looked at the file without the diff and the logic makes sense to me.

Reply:
> Why would it always be true? 🤔

`hasContext=true` only in line 31, and by that point, we know that `includeRemoteVersion` is true